src/math/warpExposure.cc

```diff
@@ -287,8 +287,8 @@ template <typename DestImageT, typename SrcImageT>
 int warpImage(DestImageT &destImage, SrcImageT const &srcImage,
               geom::TransformPoint2ToPoint2 const &srcToDest, WarpingControl const &control,
               typename DestImageT::SinglePixel padValue) {
-    if (details::isSameObject(destImage, srcImage)) {
-        throw LSST_EXCEPT(pexExcept::InvalidParameterError, "destImage is srcImage; cannot warp in place");
+    if (imagesOverlap(destImage, srcImage)) {
+        throw LSST_EXCEPT(pexExcept::InvalidParameterError, "destImage overlaps srcImage; cannot warp");
```

Comment:
The `warpImage` function requires source and destination arguments. Thus there is always  a danger that they may overlap. The only way to prevent that would be to have a version that returned a newly allocated image and we don't have that.

I'm not sure if that answers your question.